src/test/java/org/broadinstitute/consent/http/service/dao/DatasetServiceDAOTest.java

```diff
@@ -729,6 +730,76 @@ void testDeleteStudy() throws Exception {
 
   }
 
+  @Test
+  void testExecuteUpdateDatasetWithNullName() throws Exception {
+    // This creates a study with a single dataset:
+    Study study = createStudy(List.of());
+    List<Dataset> datasets = datasetDAO.findDatasetsByIdList(study.getDatasetIds().stream().toList());
```

Comment:
A better way to convert a set to a list is `copyOf()`
```suggestion
    List<Dataset> datasets = datasetDAO.findDatasetsByIdList(List.copyOf(study.getDatasetIds());
```
this creates a read-only view of the set and doesn't actually copy any data.

Although, I think an even better fix here is to change `findDatasetsByIdList()` to accept `Collection<Integer>` instead of `List<Integer>`. The docs for `@BindList` says the parameter must be an array or an Iterable so `Collection` should work.